include/ringmesh/basic/geometry.h

```diff
@@ -246,39 +246,55 @@ namespace RINGMesh
         using Disk = Circle;
     } // namespace Geometry
 
-    struct RINGMESH_API Frame3D
+    template < index_t DIMENSION >
+    struct RINGMESH_API Frame
```

Comment:
I agree with Sonar. You have getter to axis, so axis should be private. So this should not be a struct but a class.